doc/sfputil/dump_sfp_eeprom.md

```diff
@@ -0,0 +1,181 @@
+# Dump SFP EEPROM page data in show techsupport command #
+
+## Table of Content
+
+### Revision
+
+### Scope
+
+SONiC show techsupport command provides the ability to collect system dump for debug purpose. Module EEPROM data is important information for PHY issue debugging, but it is not part of show techsupport command. This design will enhance show techsupport command to contain module EEPROM data.
+
+### Definitions/Abbreviations
+
+N/A
+
+### Overview
+
+1. A new CLI `sfputil show eeprom-hexdump-all` shall dump EEPROM pages in a single call.
+2. Show techsupport command shall be extended to call `sfputil show eeprom-hexdump-all` command to collect module EEPROM data.
+
+### Requirements
+
+1. Show techsupport command shall collect module EEPROM raw data
+2. All existing cable types shall be supported except RJ45
+3. Vendor specific implementation is not required but still possible.
+
+### Architecture Design
+
+The current architecture is not changed
+
+### High-Level Design
+
+Submodule sonic-platform-common and sonic-utilities shall be changed. By default, vendor platform API implementation is not required.
+
+#### sonic-platform-common
+
+##### sfp_base change
+
+A new API `dump_eeprom` shall be added to `sfp_base.SfpBase`. This API shall be used by `sfputil show eeprom-hexdump-all` command to collect module EEPROM data.
```

Comment:
@Junchao-Mellanox we don't need a new API. We already have existing CLI `sfputil show eeprom-hexdump` and you can specify the pages you are interested in. Probably define a list of pages for each SFP type: CMIS, SFF8436, SFF8636 ...and iterate through each page from that list and invovke existing CLI